### PR TITLE
Adding UDP server error handling.

### PR DESCRIPTION
**Pull request by nicolasnoble**
Re-creating #7179 and #7169 that got lost.

@rjshade - I think you have a few more pending, just FYI.

### CODE REVIEW

**Comment by rjshade** [comment]
Thanks @nicolasnoble - what's the right way to ensure that PRs like these don't get lost in the future?

**Comment by nicolasnoble** [comment]
Harassing people, I guess ? :)

**Comment by nicolasnoble** [comment]
Also, #7178.

File: src/core/lib/iomgr/udp_server.c [hunk 1]
```diff
@@ -217,14 +217,23 @@ static int prepare_socket(int fd, const struct sockaddr *addr,
     goto error;
   }
 
-  if (!grpc_set_socket_nonblocking(fd, 1) || !grpc_set_socket_cloexec(fd, 1)) {
-    gpr_log(GPR_ERROR, "Unable to configure socket %d: %s", fd,
-            strerror(errno));
+  if (grpc_set_socket_nonblocking(fd, 1) != GRPC_ERROR_NONE) {
+    gpr_log(GPR_ERROR, "Unable to set nonblocking %d: %s", fd, strerror(errno));
```
Comment (dgquintas) [inline src/core/lib/iomgr/udp_server.c]: You are discarding all the info returned by `grpc_set_socket_nonblocking`. Ideally, this server should do something like https://github.com/grpc/grpc/blob/master/src/core/lib/iomgr/tcp_server_posix.c#L347 

At the very least, print the contents of the `grpc_error` (see `grpc_error_string` at https://github.com/grpc/grpc/blob/master/src/core/lib/iomgr/error.h#L129).

Same for the rest of the changes.

Comment (nicolasnoble) [inline src/core/lib/iomgr/udp_server.c]: You should rather have that comment on #7169 for @rjshade to take action with it.

Comment (rjshade) [inline src/core/lib/iomgr/udp_server.c]: @dgquintas thanks, these changes are intended to switch over to use grpc_error as simply as possible. I am working on a followup that will change udp_server.c to be more similar to tcp_server_posix.c in terms of error handling.

Also this is a step to fixing https://github.com/grpc/grpc/issues/6624 ... but this stuff needs to build first.

Comment (dgquintas) [inline src/core/lib/iomgr/udp_server.c]: alright :) LGTM

**Comment by rjshade** [comment]
Ping! Can this be merged? I don't want the udp_server changes to get lost again.

**Pull request merged**